src/libpgmoneta/configuration.c

```diff
@@ -3021,3 +3041,48 @@ is_empty_string(char* s)
 
    return true;
 }
+
+static int
+remove_leading_whitespace(const char* s, char** trimmed_line)
+{
+   // Find the index of the first non-whitespace character
+   int i = 0;
+   while (s[i] != '\0' && isspace(s[i]))
+   {
+      i++;
+   }
+
+   *trimmed_line = NULL; // Initialize trimmed_line
+
+   // Loop through the string starting from non-whitespace character
+   for (; s[i] != '\0'; i++)
+   {
+      *trimmed_line = pgmoneta_append_char(*trimmed_line, s[i]);
+   }
+
+   if (*trimmed_line == NULL)
+   {
+      return 1;   // Memory allocation failed
+   }
+
+   return 0;  // Return 1 if whitespace was removed
+}
+
+static void
+remove_comments(char* s)
```

Comment:
Probably best to combined the functions such that you only have one function that trimmed the line, and removes comments

Reply:
I have combined the two functions. It didn't feel right to combine while and for loop into one loop. I made all the other necessary changes.